File: HomeWork/Home4.1/PizzaDirector.java
import java.util.Scanner;

public class PizzaDirector {
    public void constructPizza(PizzaBuilder builder) {
        Scanner scanner = new Scanner(System.in);
        System.out.println("Enter ingredients (type 'done' to finish):");

        while (true) {
            String input = scanner.nextLine();
            if (input.equals("done")) {
                break;
            }
            try {
                builder.addIngredient(input);
                System.out.println(input + " added.");
            } catch (IllegalArgumentException e) {
                System.out.println(e.getMessage());
            }
        }
    }
}

File: Class/practice4.1/PizzaDirector.cpp
#include "PizzaDirector.h"
#include <iostream>
#include <format> // For std::format and std::print

// Define a named constant for the command
constexpr auto DONE_COMMAND = "done";

void PizzaDirector::constructPizza(PizzaBuilder& builder) {
    std::string input;
    std::print("Enter ingredients (type '{}' to finish):\n", DONE_COMMAND);

    while (true) {
        std::cin >> input;
        if (input == DONE_COMMAND) {
            break;
        }
        try {
            builder.addIngredient(input);
            std::print("{} added.\n", input);
        } catch (const std::exception& e) {
            std::print("{}\n", e.what());
        }
    }
}
